### PR TITLE
Better criterion for resimplify

### PR DESCRIPTION
**Pull request by Gbury**
# TL; dr
Use a more precise criterion for when to "resimplify" a function's body to avoid useless and uneeded resimplifications.

# Long explanation

As of now (i.e. without/before this PR), we sometime "resimplify" a function's code. This happens when we have done some mutable unboxing, so that we can do a second round of unboxing, particularly since mutable unboxing is done after the regular unboxing, and thus to completely unbox a ref containing a float, we unbox the ref in the first pass, and then "resimplify" to unbox the float.

Mutable unboxing can also unbox immutable blocks. This can be sometimes useful since there are some cases involving loops where mutable unboxing can do more than the regular continuation parameter unboxing[1]. Unfortunately, that means that the mutable unboxing can sometimes redo a bit of the work done by the continuation param unboxing. In particular, when only the unboxed version of a contination param is used and the allocation can be removed (which would happen on the upwards pass), the mutable unboxing pass will be able to see that and include the removal of the allocations in its result. This is a bit redundant, but not a problem in general.

To implement the "whether to resimplify ?" part, the current code looks at whether the mutable unboxing pass "did something", which translates to "is the rewrite map in the ouptut of the mutable unboxing empty or not". However, when the regular continuation param unboxing works and would lead to removal of allocations (during the upwards pass), as explained above, the rewrite map will not be empty, even though the mutable unboxing did not actually add anything (the allocation would have been removed anyway during the upwards pass, thanks to the free_names check). 

Therefore, this PR implements a better criterion for "whether the mutable unboxing did any interesting work", by ignoring the `Remove_prim` named rewrites generated by mutable unboxing.

[1]: this can happen because the continuation param unboxing for recursive continuations uses the kinds to compute the unboxing decisions, whereas the mutable unboxing looks at the allocations (i.e. the make_blocks).

### CODE REVIEW

**Review by chambart — APPROVED** [review]
This is good

**Review by Ekdohibs — APPROVED** [review]
